### ISSUE TITLE
Search settings: Verbiage and UX enhancements

### ISSUE THREAD

**Issue by jeffgolenski** [Automattic/jetpack#9747]
**Current:**

Jetpack search is a professional plan only feature, so we'll need to make adjustments for various plan levels.

**Free, personal, or premium plan enabled:**
<img width="786" alt="screen shot 2018-06-13 at 12 16 21 pm" src="https://user-images.githubusercontent.com/214813/41366469-3336446c-6f0a-11e8-8435-0a61b84ad3af.png">

**Pro plan enabled:**
<img width="762" alt="screen shot 2018-06-13 at 12 24 54 pm" src="https://user-images.githubusercontent.com/214813/41366472-34c98460-6f0a-11e8-8a00-8f45370f993e.png">

**Proposed:**

I'd like to add the description area to every plan level. And for those plans that don't offer search (free, personal, and premium), I think if we change the description of the upgrade prompt, we may do some good to help understanding.

**Free, personal, or premium plan enabled:**
<img width="766" alt="screen shot 2018-06-13 at 1 01 07 pm" src="https://user-images.githubusercontent.com/214813/41366688-eb67f6d4-6f0a-11e8-9a29-fcb6d211d226.png">

upgrade prompt description: 

> Get a faster and more powerful site-wide search with Jetpack Professional. Learn more

**Pro plan enabled:**
<img width="752" alt="screen shot 2018-06-13 at 12 54 08 pm" src="https://user-images.githubusercontent.com/214813/41366694-eeb7ab40-6f0a-11e8-9fcf-b98661438ebd.png">

**for all plan levels**

description: 

> The built-in WordPress search can slow down when you begin to gather a lot of content. Jetpack Search replaces the default search experience with an Elasticsearch-powered service made to scale as your site gets bigger. No matter how much content you have, your search results will appear almost instantly.

<img width="320" alt="screen shot 2018-06-13 at 12 25 36 pm" src="https://user-images.githubusercontent.com/214813/41366908-941886d6-6f0b-11e8-9e63-93493a62799f.png">

We'll remove the descriptive text here in the info bubble

**Comment by gibrown** [comment]
+1000 been thinking we should adjust the wording of these. I think we should do it in conjunction with changes to the Calypso language we use also: https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/site-settings/search.jsx (and I want to add upgrade prompts there also). The prompts there are both for WP.com and Jetpack.

Some thoughts:
- What about specifically branding it everywhere as "Jetpack Search" in the title of the section? I did this in Calypso, and like that it ties things together. Maybe not necessary here though?
- I think we should remove the "Elasticsearch-powered" bits. Most folks don't know or care, they just want better search. The ones who do already will know.
- I like the focus on speed in the new upgrade message
- Can we also add something about filtering options?
- Should we think about how we will add other important features here? Spelling correction for instance (not shipped yet, but we have a PR)?
- Instead of "improved search experience" I've been thinking we should switch to the phrase "advanced search experience". Feels a bit more compelling to me.

**Comment by gravityrail** [comment]
Hm. I think we do need to include a mention of search relevance, and it feels a bit wordy. Here's a little rewrite, with consideration of @gibrown's feedback above.

> WordPress' built-in search is fine for small sites, but becomes slow and inefficient as they grow.
> 
> Jetpack Search replaces the built-in search with a fast, scalable, customizable and highly relevant search engine that is hosted in the WordPress.com cloud. The result: Your users find the content they want, faster.

**Comment by gibrown** [comment]
I like that a lot.

**Comment by jeffgolenski** [comment]
@gibrown @gravityrail I like that a lot as well. 🚢 it.

It's worth mentioning that this issue is part of a larger hackweek project that stalled a few months ago: https://github.com/orgs/Automattic/projects/28#card-10550204
